app/configurator/components/chart-controls/color-ramp.tsx

```diff
@@ -1,34 +1,209 @@
+import { get } from "lodash";
+import { useEffect, useMemo, useRef } from "react";
+import { Box, Button, Text } from "theme-ui";
+import {
+  DivergingPaletteType,
+  SequentialPaletteType,
+  useConfiguratorState,
+} from "../..";
+import { Label } from "../../../components/form";
+import { Icon } from "../../../icons";
+import { divergingPalettes, sequentialPalettes } from "../ui-helpers";
 // Adapted from https://observablehq.com/@mbostock/color-ramp
-export const ColorRamp = ({
-  colorInterpolator = interpolateOranges,
-  nbClass = 512,
-  width = 288,
-  height = 40,
-}: {
+
+type ColorRampProps = {
   colorInterpolator: (t: number) => string;
-  nbClass: number;
+  nbClass?: number;
   width?: number;
   height?: number;
-}) => {
+};
+
+export const ColorRamp = ({
+  colorInterpolator,
+  nbClass = 512,
+  width = 148,
+  height = 28,
+}: ColorRampProps) => {
   const canvasRef = useRef<HTMLCanvasElement>(null);
-  // Clear by returning a function from usEEFFECT?
+
   useEffect(() => {
     const canvas = canvasRef.current;
     const context = canvas && canvas.getContext("2d");
+
     if (canvas && context) {
-      canvas.style.width = `${width}px`;
-      canvas.style.height = `${height}px`;
+      context.clearRect(0, 0, width, height);
       canvas.style.imageRendering = "-moz-crisp-edges";
       canvas.style.imageRendering = "pixelated";
-      context.clearRect(0, 0, width, height);
-      for (let i = 0; i < nbClass; ++i) {
-        context.fillStyle = colorInterpolator(i / (nbClass - 1));
-        context.fillRect((width / nbClass) * i, 0, width / nbClass, height);
+      canvas.style.borderRadius = "2px";
+
+      const [widthPerClass, numberOfSteps] =
+        nbClass > width ? [1, width] : [width / nbClass, nbClass];
+
+      for (let i = 0; i < numberOfSteps; ++i) {
+        context.fillStyle = colorInterpolator(i / (numberOfSteps - 1));
+        context.fillRect(widthPerClass * i, 0, widthPerClass, height);
       }
     }
+  }, [colorInterpolator, height, width, nbClass]);
+
+  return <canvas ref={canvasRef} width={width} height={height} />;
+};
+
+type ColorRampFieldProps = Omit<ColorRampProps, "colorInterpolator"> & {
+  field: string;
+  path: string;
+};
+
+export const ColorRampField = ({
+  field,
+  path,
+  nbClass,
+}: ColorRampFieldProps) => {
+  const [state, dispatch] = useConfiguratorState();
+
+  const palettes = useMemo(
+    () => [...divergingPalettes, ...sequentialPalettes],
+    []
+  );
+
+  const currentPaletteName = get(
+    state,
+    `chartConfig.fields.${field}.${path}`
+  ) as DivergingPaletteType | SequentialPaletteType;
+
+  const currentPalette =
+    palettes.find((d) => d.value === currentPaletteName) || palettes[0];
+
+  const {
+    isOpen,
+    getToggleButtonProps,
+    getLabelProps,
+    getMenuProps,
+    highlightedIndex,
+    getItemProps,
+  } = useSelect({
+    items: palettes,
+    defaultSelectedItem: palettes.find((d) => d.value === "oranges"),
```

Comment:
This could be extracted as a const right ? (I am a bit triggered by `find`s in render 😅 )